tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .next' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The next month is set correctly'
+    );
+
+    this.$( '.sl-calendar .prev' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The next month is set correctly'
+    );
+
+    this.$( '.sl-calendar .datepicker-switch' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The next month is set correctly'
+    );
+});
+
+test( 'Navigating from Month to Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'September ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .datepicker-switch' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+});
+
+test( 'Navigating from Year to Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+            viewMode="months"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+
+    this.$( '.sl-calendar .active' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'September ' + currentYear,
+        'The current month is set correctly'
+    );
+});
+
+test( 'Navigating from Year to Decade to Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
```

Comment:
content=content is not needed for this test and has been removed

currentMonth is also not needed and has been removed